### PR TITLE
[TR3FIR-275] update embassy

### PR DESCRIPTION
**Pull request by mattcaron-token**
The commits tagged with the ticket number should be reviewed.

The first one is a big merge commit. There were some conflicts, but I generally resolved them by keeping upstream, except when it was something we specifically added.

In later commits, I proceed to back out changes we made to upstream because we're not publishing it to artifactory anymore.

### CODE REVIEW

**Review by RafalUlko — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Comment by mattcaron-token** [comment]
I'm not certain I trust this merge quite yet (I don't want CI to overwrite old packages right on the eve of release). It **shouldn't** matter, but I don't need it merged right now, and per @alessandro-intent , we're holding off until next week, so I'm going to put this in draft until then.

Once we do the release, I'll merge this and, theoretically, nothing should happen because we're using a submodule, but I'd rather not ruin my Thursday.

**Comment by mattcaron-token** [comment]
The release was released, so I'm going to merge this and we'll see what breaks.

**Pull request merged**
